test/common/packagelib.py

```diff
@@ -87,6 +88,13 @@ def setUp(self):
             self.machine.execute(f"mkdir -p {empty_repo_dir} || true")
             self.machine.execute(f"repo-add {empty_repo_dir}/empty.db.tar.gz")
             self.machine.write("/etc/pacman.conf", config)
+            # Clean up possible leftover lockfile
+            self.machine.execute("""
+            if [ -f /var/lib/pacman/db.lck ]; then
+                fuser -k /var/lib/pacman/db.lck && sleep 0.2 || true;
```

Comment:
hmm, sleep -- can this be done with a while loop instead? so that it times out when it's unable to kill the process, and we get better test failure feedback?